app/lib/application_logger.rb

```diff
@@ -0,0 +1,43 @@
+require "json"
+
+class ApplicationLogger < ActiveSupport::Logger
```

Comment:
Minor thing but worth doing: you could split adding this class and the spec in a prior commit to make this one a smaller set of changes in flight

Reply:
I've split this out in the runner PR as there are more changes so thought it was worthwhile. While I agree with the principle, can I gently push back on doing that here, because I think the changes make more sense when viewed all together and also while its not much effort, it is effort unpicking and redoing commits.

Reply:
"Do the hard work to make it simple" - I understand the effort, and that's why I suggested that video 😄 

It's literally much harder for me to understand.

Reply:
For further context and for others to understand where I'm coming from: my adhd couldn’t cope - and it’s not just for me but for any other folks who struggle as I do. Generally unless it’s a “find and replace all” type of change, if the number of files affected is more than ~5 or 6 files I collapse under the weight of the page scroll (see also very large diffs).

This is the video I mentioned: https://www.youtube.com/watch?v=GyUxe7PMD4A

Reply:
I've separated this out now. Thanks for the context on why it's useful to do this!

Reply:
TYSM!